pyedflib/edfreader.py

```diff
@@ -59,60 +68,84 @@ def _debug_parse_header(filename): #pragma: no cover
         prefilter = [f.read(80).decode().strip() for i in range(nsigs)]
         n_samples = [f.read(8).decode() for i in range(nsigs)]
         reserved = [f.read(32).decode() for i in range(nsigs)]
-    _ = zip(label, transducer, dimension, pmin, pmax, dmin, dmax, prefilter, n_samples, reserved)
+    _ = zip(
+        label,
+        transducer,
+        dimension,
+        pmin,
+        pmax,
+        dmin,
+        dmax,
+        prefilter,
+        n_samples,
+        reserved,
+    )
     values = locals().copy()
-    fields = ['label', 'transducer', 'dimension', 'pmin', 'pmax', 'dmin', 'dmax', 'prefilter', 'n_samples', 'reserved']
-    sheaders = [{field:values[field][i] for field in fields} for i in range(nsigs)]
-    print('\n##### Signal Headers')
+    fields = [
+        "label",
+        "transducer",
+        "dimension",
+        "pmin",
+        "pmax",
+        "dmin",
+        "dmax",
+        "prefilter",
+        "n_samples",
+        "reserved",
+    ]
+    sheaders = [{field: values[field][i] for field in fields} for i in range(nsigs)]
+    print("\n##### Signal Headers")
     print(json.dumps(sheaders, indent=2))
 
 
 class EdfReader(CyEdfReader):
     """
     This provides a simple interface to read EDF, EDF+, BDF and BDF+ files.
     """
-    def __enter__(self):
+
+    def __enter__(self) -> "EdfReader":
         return self
 
-    def __del__(self):
+    def __del__(self) -> None:
         self._close()
 
-    def __exit__(self, exc_type, exc_val, ex_tb):
+    def __exit__(self, exc_type, exc_val, ex_tb) -> None:
```

Comment:
```suggestion
def __exit__(
    self,
    exc_type: Optional[Type[BaseException]],
    exc_val: Optional[BaseException],
    exc_tb: Optional[TracebackType],
) -> None:
```

see comment below, need to import tracebacktype as well